src/NuGet.Services.AzureSearch/Catalog2AzureSearch/DocumentFixUp.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.Services.Metadata.Catalog;
+
+namespace NuGet.Services.AzureSearch.Catalog2AzureSearch
+{
+    public class DocumentFixUp
+    {
+        private DocumentFixUp(bool applicable, List<CatalogCommitItem> itemList)
+        {
+            Applicable = applicable;
+            ItemList = itemList;
+        }
+
+        public static DocumentFixUp IsNotApplicable()
+        {
+            return new DocumentFixUp(applicable: false, itemList: null);
+        }
+
+        public static DocumentFixUp IsApplicable(List<CatalogCommitItem> itemList)
+        {
+            return new DocumentFixUp(applicable: true, itemList: itemList);
```

Comment:
Super nit: should we validate that `itemList` is non-null and non-empty?

Reply:
Sure, sounds fine